library/std/src/sync/lazy_lock.rs

```diff
@@ -69,6 +69,16 @@ impl<T, F: FnOnce() -> T> LazyLock<T, F> {
         LazyLock { once: Once::new(), data: UnsafeCell::new(Data { f: ManuallyDrop::new(f) }) }
     }
 
+    /// Creates a new lazy value that is already initialized.
+    #[inline]
+    #[cfg_attr(not(test), allow(unused))] // Only used in tests.
```

Comment:
It would have to be applied to all internal functions as well, so this seemed simpler...

Reply:
`completed` are pub though, so it shouldn't get linted, I think? 🤔

Reply:
Actually, I think instead of doing `Once::completed()` you can do
```rust
let once = Once::new();
once.call_once(|| {});
```

This has an added advantage of not requiring the addition of a new constructor.

This will lose the `const`ness, but I think for tests it should be fine.

Reply:
> Actually, I think instead of doing `Once::completed()` you can do
> 
> ```rust
> let once = Once::new();
> once.call_once(|| {});
> ```
> 
> This has an added advantage of not requiring the addition of a new constructor.
> 
> This will lose the `const`ness, but I think for tests it should be fine.

True. It's inefficient, but definitely simpler.

Reply:
Since this is for a couple of tests, I don't think efficiency matters much.

Reply:
Changed it. Since it does not call any other new functions, it uses `#[cfg(test)]` now.